sale_order_import_edifact/wizard/sale_order_import.py

```diff
@@ -0,0 +1,237 @@
+import json
+import logging
+import os
+from base64 import b64decode
+from collections import defaultdict
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+logger = logging.getLogger(__name__)
+
+
+class SaleOrderImportEdifact(models.TransientModel):
+    _name = "sale.order.import"
+    _inherit = ["sale.order.import", "base.edifact"]
+
+    state = fields.Selection(
+        selection_add=[("tech", "Technical")], ondelete={"tech": "set default"}
+    )
+    parsed_json = fields.Text()
+
+    @property
+    def edifact_ok(self):
+        conf_ext = self.env.context.get("file_ext", "txt,d96a")
+        extensions = conf_ext.split(",")
+        ok = False
+        if self.order_filename:
+            path, ext = os.path.splitext(self.order_filename)
+            ok = ext and ext[1:] in extensions
+            if not ok and self.order_file:
+                ok = b64decode(self.order_file[:4]) == b"UNB"
+        return ok
+
+    @api.onchange("order_file")
+    def order_file_change(self):
+        if self.edifact_ok:
```

Comment:
see my comments on https://github.com/OCA/edi/pull/816

Reply:
I don't understand this. We should leverage the call to super for loading defaults.
Also, here, the doc_type and the price_source are kind of hardcoded...
Shouldn't we rely on the parsing of the file as for the other formats?

Reply:
This is to be able to force from configuration.

```yaml
sale_order_import:
      price_source: order
      confirm_order: false
      wiz_ctx:
         file_ext: 'edi'
         release: 'd96a'
         doc_type: 'rfq'
```
but yes, `price_source` should be taken from context like `doc_type`

Reply:
that's the point. Instead of having special keys, we can rely on the fact that defaults will be taken automatically.
In they yml example above I'd like to get rid of the special keys `price_source` and `confirm_order`.
In the wiz ctx we should see something like:

```
wiz_ctx:
    default_file_ext: edi
    default_release: d96a
    default_doc_type: rfq
    default_price_source: order
    default_confirm_order: true
```
and if I'm not mistaken the only field missing here is `confirm_order` which I'd like to add as a boolean in the sale.order.import wizard to automatically confirm it instead for relying on a custom machinery in edi_sale_oca.
TBH is my "fault". Is something that I didn't introduce in v 14 and now I'd like to clear it up.
Hope is clear :wink:

Reply:
I created a PR for `confirm_order` field in `sale_order_import` module at https://github.com/OCA/edi/pull/839
Could you have a look?

Reply:
FTR https://github.com/OCA/edi-framework/pull/31#discussion_r1372739178